google/resource_compute_instance.go

```diff
@@ -1142,3 +1235,79 @@ func resourceInstanceTags(d *schema.ResourceData) *compute.Tags {
 
 	return tags
 }
+
+func expandBootDisk(d *schema.ResourceData, config *Config, zone *compute.Zone, project string) (*compute.AttachedDisk, error) {
+	disk := &compute.AttachedDisk{
+		AutoDelete: d.Get("boot_disk.0.auto_delete").(bool),
+		Boot:       true,
+	}
+
+	if v, ok := d.GetOk("boot_disk.0.device_name"); ok {
+		disk.DeviceName = v.(string)
+	}
+
+	if v, ok := d.GetOk("boot_disk.0.disk_encryption_key_raw"); ok {
+		disk.DiskEncryptionKey = &compute.CustomerEncryptionKey{
+			RawKey: v.(string),
+		}
+	}
+
+	if v, ok := d.GetOk("boot_disk.0.source"); ok {
+		diskName := v.(string)
+		diskData, err := config.clientCompute.Disks.Get(
+			project, zone.Name, diskName).Do()
+		if err != nil {
+			return nil, fmt.Errorf("Error loading disk '%s': %s", diskName, err)
+		}
+		disk.Source = diskData.SelfLink
+	}
+
+	if _, ok := d.GetOk("boot_disk.0.initialize_params"); ok {
+		disk.InitializeParams = &compute.AttachedDiskInitializeParams{}
+
+		if v, ok := d.GetOk("boot_disk.0.initialize_params.0.size"); ok {
+			disk.InitializeParams.DiskSizeGb = int64(v.(int))
+		}
+
+		if v, ok := d.GetOk("boot_disk.0.initialize_params.0.type"); ok {
+			diskTypeName := v.(string)
+			diskType, err := readDiskType(config, zone, diskTypeName)
+			if err != nil {
+				return nil, fmt.Errorf("Error loading disk type '%s': %s", diskTypeName, err)
+			}
+			disk.InitializeParams.DiskType = diskType.Name
+		}
+
+		if v, ok := d.GetOk("boot_disk.0.initialize_params.0.image"); ok {
+			imageName := v.(string)
+			imageUrl, err := resolveImage(config, imageName)
+			if err != nil {
+				return nil, fmt.Errorf("Error resolving image name '%s': %s", imageName, err)
+			}
+
+			disk.InitializeParams.SourceImage = imageUrl
+		}
+	}
+
+	return disk, nil
+}
+
+func flattenBootDisk(d *schema.ResourceData, disk *compute.AttachedDisk) []map[string]interface{} {
```

Comment:
Same as expandBootDisk - do we need access to all of `schema.ResourceData` here? Or just a subset?